### PR TITLE
Track CC, CD, DC, DD in Player class

### PR DESCRIPTION
**Pull request by mturzanska**
Related to https://github.com/Axelrod-Python/Axelrod/issues/744

Add play_counts property to Player class (a dictionary counting occurrences of CC, CD, DC, DD).
Update the dictionary after each play.
Add tests for the property.

TODO:
Use play_counts property in existing strategies (e.g. Prober4, Retaliate)

### CODE REVIEW

File: axelrod/tests/unit/test_player.py [hunk 1]
```diff
@@ -60,6 +64,9 @@ def test_play(self):
         self.assertEqual(p1.defections, 0)
         self.assertEqual(p2.cooperations, 0)
         self.assertEqual(p2.defections, 2)
+        # Test play counts
+        self.assertEqual(p1.play_counts, {(C, D): 2})
+        self.assertEqual(p2.play_counts, {(D, C): 2})
```
Comment (drvinceknight) [inline axelrod/tests/unit/test_player.py]: Could we have a couple more tests leading to a 'fuller' dictionary please.

File: axelrod/player.py [hunk 2]
```diff
@@ -52,6 +53,12 @@ def update_history(player, move):
         player.defections += 1
 
 
+def update_play_counts(player, move, reply):
```
Comment (drvinceknight) [inline axelrod/player.py]: In the glossary <http://axelrod.readthedocs.io/en/latest/reference/glossary.html#a-play> we have defined a play as "a single player choosing an action."

In the results set and the interaction utils we have called these counters "state distributions". Could we call them the same thing here (for consistency).

- `update_state_distribution`
- `player_state_distribution`

Comment (drvinceknight) [inline axelrod/player.py]: Yup: that's better than my suggestion :) 👍

**Comment by marcharper** [comment]
Since we're counting C and D as well for each player (in attributes cooperations and defections), should we change those to a dictionary as well? (Not necessarily in this PR -- there will be a lot of places that need to change.)

**Comment by drvinceknight** [comment]
> Since we're counting C and D as well for each player (in attributes cooperations and defections), should we change those to a dictionary as well? (Not necessarily in this PR -- there will be a lot of places that need to change.)

Good idea but yeah: this could/should be another issue.

**Comment by drvinceknight** [comment]
Thanks for the changes @mturzanska, this looks good to me now.

**Pull request merged**